staging/src/k8s.io/apiextensions-apiserver/test/integration/registration_test.go

```diff
@@ -42,30 +42,7 @@ import (
 )
 
 func instantiateCustomResource(t *testing.T, instanceToCreate *unstructured.Unstructured, client dynamic.ResourceInterface, definition *apiextensionsv1beta1.CustomResourceDefinition) (*unstructured.Unstructured, error) {
-	createdInstance, err := client.Create(instanceToCreate)
-	if err != nil {
-		t.Logf("%#v", createdInstance)
-		return nil, err
-	}
-	createdObjectMeta, err := meta.Accessor(createdInstance)
-	if err != nil {
-		t.Fatal(err)
-	}
-	// it should have a UUID
-	if len(createdObjectMeta.GetUID()) == 0 {
-		t.Errorf("missing uuid: %#v", createdInstance)
-	}
-	createdTypeMeta, err := meta.TypeAccessor(createdInstance)
-	if err != nil {
-		t.Fatal(err)
-	}
-	if e, a := definition.Spec.Group+"/"+definition.Spec.Version, createdTypeMeta.GetAPIVersion(); e != a {
-		t.Errorf("expected %v, got %v", e, a)
-	}
-	if e, a := definition.Spec.Names.Kind, createdTypeMeta.GetKind(); e != a {
-		t.Errorf("expected %v, got %v", e, a)
-	}
-	return createdInstance, nil
+	return instantiateVersionedCustomResource(t, instanceToCreate, client, definition, definition.Spec.Versions[0].Name)
```

Comment:
should we call it `instantiateCustomResource` everywhere?

Reply:
nit: (also previous one is nit) or we can call it the versioned but but I think we should just have one name for it.

Reply:
had that before, but I don't want to add references to `Versions[0]` everywhere. That's becoming messy.

Reply:
Let's turn that around: rename `instantiateCustomResource` to `instantiateV1beta1CustomResource`

Reply:
Leaving it as it is for now. Cannot come up with a less ugly variant.